src/Microsoft.Identity.Web/ClientInfo.cs

```diff
@@ -25,19 +25,11 @@ public static ClientInfo CreateFromJson(string clientInfo)
                 throw new ArgumentNullException(nameof(clientInfo), $"client info returned from the server is null");
             }
 
-            return DeserializeFromJson<ClientInfo>(Base64UrlHelpers.DecodeToBytes(clientInfo));
-        }
-
-        internal static T DeserializeFromJson<T>(byte[] jsonByteArray)
```

Comment:
I would keep `DeserializeFromJson`. I think you can also leave the generic parameter and add `where T : notnull`.